shpec/build_shpec.sh

```diff
@@ -55,4 +55,20 @@ describe "lib/build.sh"
     end
   end
 
+  describe "check_tsc"
+    it "runs tsc if no out_dir is detected"
+      project_dir=$(create_temp_project_dir)
+
+      set +e
+      check_tsc "$project_dir"
+      loc_var=$?
+      set -e
+
+      out_dir_check=$(detect_out_dir "$project_dir")
+
+      assert equal "$loc_var" 0
+      assert equal "$out_dir_check" 0
```

Comment:
For this test, we can look an absent file, then run the method to compile, then check to see if the file created the outDir. There's are shpec helpers that will look for folders: https://github.com/rylnd/shpec#unary-matchers (we want to use `file_present` and `file_absent`)